### PR TITLE
added code in C and C++ for Bezier Curve

### LINKED ISSUE #1028: Beizer curve
Implementation of beizer curve

### PR DESCRIPTION
**Pull request by tusharsabhani**
Fixes #1028

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added code in C and C++ for Bezier Curve

#### Languages Used:
- C
- C++

#### Files Added:
- bezier_curve.c 
- bezier_curve.cpp

### CODE REVIEW

**Comment by KanchanThareja** [comment]
@tusharsabhani please mention issue number and fix errors in your pr description.

**Comment by tusharsabhani** [comment]
What errors?

**Comment by tusharsabhani** [comment]
Please check once

File: Bezier Curve/bezier_curve.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    double t;
+    initgraph (&gd, &gm, "");
+
+    for (t = 0.0; t < 1.0; t += 0.0005)
+    {
+	    double xt = pow (1-t, 3) * x[0] + 3 * t * pow (1-t, 2) * x[1] + 3 * pow (t, 2) * (1-t) * x[2] + pow (t, 3) * x[3];
+	    double yt = pow (1-t, 3) * y[0] + 3 * t * pow (1-t, 2) * y[1] + 3 * pow (t, 2) * (1-t) * y[2] + pow (t, 3) * y[3];
+	    putpixel (xt, yt, WHITE);
+    }
+    for (i=0; i<4; i++)
```
Comment (somya-kapoor) [inline Bezier Curve/bezier_curve.c]: Please add space before and after =,<.

File: Bezier Curve/bezier_curve.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+//function that takes coordinates of control points as input and then draws the curve
+void bezier (int x[4], int y[4])
+{
+    int gd = DETECT, gm;
+    initgraph (&gd, &gm, "");
+
+    for (double t = 0.0; t < 1.0; t += 0.0005)
+    {
+	    double xt = pow (1-t, 3) * x[0] + 3 * t * pow (1-t, 2) * x[1] + 3 * pow (t, 2) * (1-t) * x[2] + pow (t, 3) * x[3];
```
Comment (somya-kapoor) [inline Bezier Curve/bezier_curve.cpp]: Please fix the indentation ( 4 spaces)

File: Bezier Curve/bezier_curve.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+void bezier (int x[4], int y[4])
+{
+    int gd = DETECT, gm;
+    int i;
+    double t;
+    initgraph (&gd, &gm, "");
+
+    for (t = 0.0; t < 1.0; t += 0.0005)
+    {
+	    double xt = pow (1-t, 3) * x[0] + 3 * t * pow (1-t, 2) * x[1] + 3 * pow (t, 2) * (1-t) * x[2] + pow (t, 3) * x[3];
```
Comment (somya-kapoor) [inline Bezier Curve/bezier_curve.c]: Please fix the indentation ( 4 spaces )

File: Bezier Curve/bezier_curve.cpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	    double yt = pow (1-t, 3) * y[0] + 3 * t * pow (1-t, 2) * y[1] + 3 * pow (t, 2) * (1-t) * y[2] + pow (t, 3) * y[3];
+	    putpixel (xt, yt, WHITE);
+    }
+    for (int i=0; i<4; i++)
+	    putpixel (x[i], y[i], YELLOW);
```
Comment (somya-kapoor) [inline Bezier Curve/bezier_curve.cpp]: Please fix the indentation ( 4 spaces)

File: Bezier Curve/bezier_curve.cpp [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    int gd = DETECT, gm;
+    initgraph (&gd, &gm, "");
+
+    for (double t = 0.0; t < 1.0; t += 0.0005)
+    {
+	    double xt = pow (1-t, 3) * x[0] + 3 * t * pow (1-t, 2) * x[1] + 3 * pow (t, 2) * (1-t) * x[2] + pow (t, 3) * x[3];
+	    double yt = pow (1-t, 3) * y[0] + 3 * t * pow (1-t, 2) * y[1] + 3 * pow (t, 2) * (1-t) * y[2] + pow (t, 3) * y[3];
+	    putpixel (xt, yt, WHITE);
+    }
+    for (int i=0; i<4; i++)
```
Comment (somya-kapoor) [inline Bezier Curve/bezier_curve.cpp]: Please add space before and after =,<.

**Comment by somya-kapoor** [comment]
@tusharsabhani Please fix the file name.  ( Bezier_Curve/Bezier_Curve.c)

**Comment by tusharsabhani** [comment]
Fixed spacing

**Comment by somya-kapoor** [comment]
@tusharsabhani Please fix the file name.

**Comment by tusharsabhani** [comment]
Please have a look now

File: Bezier_Curve/Bezier_Curve.c.c [hunk 6]
```diff
@@ -0,0 +1,35 @@
+#include <stdlib.h>
+#include <graphics.h>
+#include <math.h>
+#include<iostream.h>
```
Comment (somya-kapoor) [inline Bezier_Curve/Bezier_Curve.c.c]: Please add space after include

File: Bezier_Curve/Bezier_Curve.cpp.cpp [hunk 7]
```diff
@@ -0,0 +1,32 @@
+#include <stdlib.h>
+#include <graphics.h>
+#include <math.h>
+#include<iostream.h>
```
Comment (somya-kapoor) [inline Bezier_Curve/Bezier_Curve.cpp.cpp]: Please add space after include

**Comment by somya-kapoor** [comment]
@tusharsabhani Good work 
After doing the changes, please squash the commits.

**Comment by tusharsabhani** [comment]
Squash?

**Comment by somya-kapoor** [comment]
Have a look at https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git.
If you still face issue. We can discuss on slack.

**Comment by tusharsabhani** [comment]
check now once

**Comment by jainaman224** [comment]
@tusharsabhani Please squash your commits and it is ready to be merged.

**Comment by tusharsabhani** [comment]
Check once I have squashed all my commits

**Pull request merged**